server/app/services/Path.java

```diff
@@ -100,24 +102,47 @@ public Path replacingLastSegment(String finalSegment) {
   }
 
   /**
-   * Append a path to the path.
+   * Append a String path to the path.
    *
    * <p>If joining a {@link Scalar}, please use {@link Path#join(Scalar)} instead.
+   *
+   * @param path the path in String form to join with this path
+   * @return this path joined with the provided path
    */
   public Path join(String path) {
     Path other = Path.create(path);
-    return Path.create(
-        ImmutableList.<String>builder().addAll(segments()).addAll(other.segments()).build());
+    return join(other);
   }
 
   /**
-   * Append a {@link Scalar} to the path
+   * Append a {@link Scalar} to the {@link Path}.
    *
-   * <p>This is just a helper method so we don't have to use {@link Scalar#toString()} when we want
-   * to append to a path.
+   * @param scalar the Scalar to join with this path
+   * @return this path joined with the provided Scalar
    */
   public Path join(Scalar scalar) {
     Path other = Path.create(scalar.name());
+    return join(other);
+  }
+
+  /**
+   * Append a {@link ApiPathSegment} to the {@link Path}.
+   *
+   * @param apiPathSegment the ApiPathSegment to join with this path
+   * @return this path joined with the provided ApiPathSegment
+   */
+  public Path join(ApiPathSegment apiPathSegment) {
+    Path other = Path.create(apiPathSegment.name());
+    return join(other);
+  }
+
+  /**
+   * Append a {@link Path} to this {@link Path}.
```

Comment:
**nitpick:** this -> the for consistency

Reply:
Done.